pyramid_generator/image_processor.py

```diff
@@ -64,83 +65,26 @@ def _cleanup(self) -> None:
         os.remove(self.local_file)
         os.remove(self.tif_file)
 
-
-class S3ImageProcessor(ImageProcessor):
-    def __init__(self) -> None:
-        super().__init__()
-
-    def process(self) -> dict:
-        # if self._previously_processed():
-        #     self._log_result('status', 'processed')
-        #     self._log_result('reason', 'no changes to image since last run')
-        # else:
-        img_bucket, key = self.source_image.split('/', 2)[-1].split('/', 1)
-        s3_file = f"{self.img_write_base}/{self.id}/images/{self.tif_file}"
-        self.S3_RESOURCE.Bucket(img_bucket).download_file(key, self.local_file)
-        self._generate_pytiff(self.local_file, self.tif_file)
-        self.S3_RESOURCE.Bucket(self.bucket).upload_file(self.tif_file, s3_file)
-        self._cleanup()
-        self._log_result('status', 'processed')
-        return self.image_result
-
     def _previously_processed(self) -> bool:
-        # has this been ran before and has it changed?
-        previous_image = f"{self.img_write_base}/{self.id}/images/{self.filename}"
-        response = self._get_image_metadata(self.bucket, previous_image)
-        return self.source_md5sum == response['ETag']
-
-    def _get_image_metadata(self, bucket: str, key: str, **kwargs) -> dict:
+        image_modified = False
+        if self.id not in self.prior_results:
+            self.prior_results.update({self.id: {}})
+            img_data = f"{self.img_write_base}/{self.id}/image_data.json"
+            self._set_prior_run_data(self.bucket, img_data, self.id)
+        prior_md5sum = self.prior_results.get(self.id).get(self.filename, {}).get('md5sum', 'nomatch')
+        if self.source_md5sum == prior_md5sum:
```

Comment:
Should this be != ?  It seems that the image has been modified if the old checksum and the new checksum are different.

Reply:
Blah yeah had that variable name was a misnomer. I just got rid of it for clarification.